src/test/java/hudson/plugins/jira/UpdaterTest.java

```diff
@@ -352,5 +355,75 @@ public void testUserPatternMatchTwoIssuesInOneComment() {
+    	final JiraIssue firstIssue = new JiraIssue("FOOBAR-1", "Title");
+    	final JiraIssue secondIssue = new JiraIssue("ALIBA-1", "Title");
+    	final JiraIssue thirdIssue = new JiraIssue("MOONA-1", "Title");
+    			
+    	
+    	//	assume that there is a following list of jira issues from scm commit messages out of hudson.plugins.jira.JiraCarryOverAction
+    	List<JiraIssue> issues = Lists.newArrayList(firstIssue, brokenIssue, secondIssue, forbiddenIssue, thirdIssue);
+
+        // mock JIRA session:
+        JiraSession session = mock(JiraSession.class);
+        when(session.existsIssue(Mockito.anyString())).thenReturn(Boolean.TRUE);
+        when(session.getIssue(Mockito.anyString())).thenReturn(new RemoteIssue());
+        when(session.getGroup(Mockito.anyString())).thenReturn(new RemoteGroup("Software Development", null));
+
+        final List<RemoteComment> comments = new ArrayList<RemoteComment>();
+
+        Answer answer = new Answer<Object>() {
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                RemoteComment rc = new RemoteComment();
+                rc.setId((String) invocation.getArguments()[0]);
+                rc.setBody((String) invocation.getArguments()[1]);
+                rc.setGroupLevel((String) invocation.getArguments()[2]);
+                comments.add(rc);
+                return null;
+            }
+        };
+        doAnswer(answer).when(session).addComment(eq(firstIssue.id), Mockito.anyString(), Mockito.anyString(), Mockito.anyString());
+        doAnswer(answer).when(session).addComment(eq(secondIssue.id), Mockito.anyString(), Mockito.anyString(), Mockito.anyString());
+        doAnswer(answer).when(session).addComment(eq(thirdIssue.id), Mockito.anyString(), Mockito.anyString(), Mockito.anyString());
+        
+        //	issue for the catched exception
+        doThrow(new RemotePermissionException()).when(session).addComment(eq(forbiddenIssue.id), Mockito.anyString(), Mockito.anyString(), Mockito.anyString());
+        //	issue for the exception that has not been catched
+        doThrow(new RemoteException("an error occured")).when(session).addComment(eq(brokenIssue.id), Mockito.anyString(), Mockito.anyString(), Mockito.anyString());
+
+        // mock build:
+        FreeStyleBuild build = mock(FreeStyleBuild.class);
+        FreeStyleProject project = mock(FreeStyleProject.class);
+        when(build.getProject()).thenReturn(project);
+        ChangeLogSet changeLogSet = ChangeLogSet.createEmpty(build);
+        when(build.getChangeSet()).thenReturn(changeLogSet);
+        when(build.getResult()).thenReturn(Result.SUCCESS);
+        
+        final String groupVisibility = "";
+        final String roleVisibility = "";
+        
+        try {
+        	Updater.submitComments(build,
+                System.out, "http://jenkins", issues, session, false, false, groupVisibility, roleVisibility);
```

Comment:
you should add fail() after this line, because you expect to be in a catch block already